### PR TITLE
Add crab scripts, options to save provenance in postprocessing and create FJR 

### PR DESCRIPTION
**Pull request by arizzi**
Run post processing on crab.
Support lumi-splitting and eventsAwareLumiSplitting
runsAndLumis json can now be passed as a dict rather than json file

Also introduce options to create FrameworkJobReport out of the python postprocessor (used by the crab config)

### CODE REVIEW

File: python/postprocessing/framework/jobreport.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	   ET.SubElement(infile,"InputType").text="primaryFiles"
+	   ET.SubElement(infile,"ModuleLabel").text="source"
+	   ET.SubElement(infile,"InputSourceClass").text="PoolSource"
+	   ET.SubElement(infile,"GUID").text=""
+	   ET.SubElement(infile,"EventsRead").text="%s"%eventsRead
+	   runs=ET.SubElement(infile,"Runs")
+	   for r,ls in runsAndLumis.iteritems() :
+	   	run=ET.SubElement(runs,"Run",ID="%s"%r)
+	   	for l in ls :
+		   ET.SubElement(run,"LumiSection",ID="%s"%l)
```
Comment (gpetruc) [inline python/postprocessing/framework/jobreport.py]: these we could fill with the correct info easily

File: python/postprocessing/framework/jobreport.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+       def addOutputFile(self,filename,events=1,runsAndLumis={"1":[1]}):
+           infile = ET.SubElement(self.fjr, "File")
+           ET.SubElement(infile,"LFN").text=""
```
Comment (gpetruc) [inline python/postprocessing/framework/jobreport.py]: `re.sub(r".*?/(/store/.*\.root)(\?.*)?","$1", filename)` ? 
(o via edmFileUtil)

File: python/postprocessing/framework/postprocessor.py [hunk 3]
```diff
@@ -20,6 +22,12 @@ def __init__(self,outputDir,inputFiles,cut=None,branchsel=None,modules=[],compre
 	self.noOut=noOut
 	self.friend=friend
 	self.justcount=justcount
+	self.provenance=provenance
+	self.jobReport = JobReport() if fwkJobReport else None
+	self.haddFileName=haddFileName
+	if self.jobReport and not self.haddFileName :
+		print "Because you requested a FJR we assume you want the final hadd. No name specified for the output file, will use tree.root"
+		self.haddFileName="tree.root"
```
Comment (gpetruc) [inline python/postprocessing/framework/postprocessor.py]: io forse qui metterei throw

File: python/postprocessing/framework/jobreport.py (hunk 1 above)
Comment (arizzi) [reply python/postprocessing/framework/jobreport.py]: yes, also the total number of events and other info can be corrected

File: python/postprocessing/framework/jobreport.py (hunk 2 above)
Comment (arizzi) [reply python/postprocessing/framework/jobreport.py]: it doesn't matter this one, it is anyhow moved later by crab

File: python/postprocessing/framework/postprocessor.py (hunk 3 above)
Comment (arizzi) [reply python/postprocessing/framework/postprocessor.py]: actually tree.root is the name the crab config expects, so it is fine to have a (this) default

**Comment by arizzi** [comment]
should be working now

it can be merged

**Comment by gpetruc** [comment]
Will you also update the FJR in order to put meaningful info on the number of events and run/lumi content of the files, or that will come later?

**Comment by arizzi** [comment]
definitelly later...

**Pull request merged**